src/test/lit-element_test.ts

```diff
@@ -1243,26 +1310,238 @@ suite('LitElement', () => {
+    assert.equal(sub.shadowRoot!.textContent, '0');
+
+    sub.foo = 5;
+    await sub.updateComplete;
+    assert.equal(sub.foo, 5);
+    assert.equal(sub._oldFoo, 0);
+    assert.equal(sub._subSet, 2);
+    assert.equal(sub.shadowRoot!.textContent, '5');
+
+    sub.foo = 7.5;
+    await sub.updateComplete;
+    assert.equal(sub.foo, 7); // (sub setter rounds down)
+    assert.equal(sub._oldFoo, 5);
+    assert.equal(sub._subSet, 3);
+    assert.equal(sub.shadowRoot!.textContent, '7');
+
+    sub.foo = 20;
+    await sub.updateComplete;
+    assert.equal(sub.foo, 10); // (super user getter maxes at 10)
+    assert.equal(sub._oldFoo, 7);
+    assert.equal(sub._subSet, 4);
+    assert.equal(sub.shadowRoot!.textContent, '10');
+  });
+
+  test('user accessors only using noAccessor', async () => {
+    class E extends LitElement {
+      _updateCount = 0;
+      _foo?: String;
+      _bar?: String;
+      static get properties() {
+        return {
+          foo : {type : String, reflect : true, noAccessor : true},
+          bar : {type : String, reflect : true, noAccessor : true}
+        };
+      }
+      constructor() {
+        super();
+        this.foo = 'defaultFoo';
+        this.bar = 'defaultBar';
+      }
+      set foo(value: string) {
+        const old = this._foo;
+        this._foo = value;
+        this.requestUpdate('foo', old);
+      }
+      get foo() { return this._foo as string; }
+      set bar(value: string) {
+        const old = this._bar;
+        this._bar = value;
+        this.requestUpdate('bar', old);
+      }
+      get bar() { return this._bar as string; }
+      update(changedProperties: PropertyValues) {
+        this._updateCount++;
+        super.update(changedProperties);
+      }
+      render() { return html`${this.foo}-${this.bar}`; }
+    }
+    customElements.define(generateElementName(), E);
+
+    const el = new E();
+    el.foo = 'foo1';
+    document.body.appendChild(el);
+    await el.updateComplete;
+    assert.equal(el.foo, 'foo1');
+    assert.equal(el.bar, 'defaultBar');
+    assert.equal(el.getAttribute('foo'), 'foo1');
+    assert.equal(el.getAttribute('bar'), 'defaultBar');
+    assert.equal(el.shadowRoot!.textContent, 'foo1-defaultBar');
+    assert.equal(el._updateCount, 1);
+
+    el.foo = 'foo2';
+    el.bar = 'bar';
+    await el.updateComplete;
+    assert.equal(el.foo, 'foo2');
+    assert.equal(el.bar, 'bar');
+    assert.equal(el.getAttribute('foo'), 'foo2');
+    assert.equal(el.getAttribute('bar'), 'bar');
+    assert.equal(el.shadowRoot!.textContent, 'foo2-bar');
+    assert.equal(el._updateCount, 2);
+
+    el.foo = 'foo3';
+    await el.updateComplete;
+    assert.equal(el.foo, 'foo3');
+    assert.equal(el.getAttribute('foo'), 'foo3');
+    assert.equal(el.getAttribute('bar'), 'bar');
+    assert.equal(el.shadowRoot!.textContent, 'foo3-bar');
+    assert.equal(el._updateCount, 3);
+  });
+
+  test('attributeChangedCallback based rendering', async () => {
```

Comment:
Suggest duplicating this test without `noAccessor: true` and not overriding `attributeChangedCallback`.

Reply:
Done